src/Providers/Middleware/CheckClientCredentials.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace LaravelSsoClient\Providers\Middleware;
+
+use Closure;
+use LaravelSsoClient\Exceptions\MissingScopeException;
+use Illuminate\Auth\AuthenticationException;
+use LaravelSsoClient\JWT;
+
+class CheckClientCredentials
+{
+    /**
+     * JWT.
+     *
+     * @param  LaravelSsoClient\JWT  $jwt
+     */
+    protected JWT $jwt;
+
+    /**
+     * Create a new middleware instance.
+     *
+     * @param  LaravelSsoClient\JWT  $jwt
+     * @return void
+     */
+    public function __construct(JWT $jwt)
+    {
+        $this->jwt = $jwt;
+    }
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @param  mixed  ...$scopes
+     * @return mixed
+     *
+     * @throws \Illuminate\Auth\AuthenticationException
+     */
+    public function handle($request, Closure $next, ...$scopes)
+    {
+        $this->validate($scopes);
+
+        return $next($request);
+    }
+
+    /**
+     * Validate the scopes and jwt on the incoming request.
+     *
+     * @param  array  $scopes
+     * @return true
+     *
+     * @throws \LaravelSsoClient\Exceptions\|\Illuminate\Auth\AuthenticationException
+     */
+    public function validate($scopes)
+    {
+        $this->validateClaims();
+
+        $this->validateScopes($scopes);
+
+        return true;
+    }
+
+    /**
+     * Validate jwt claims.
+     *
+     * @return void
+     *
+     * @throws \Illuminate\Auth\AuthenticationException
+     */
+    protected function validateClaims()
+    {
+        if (!$this->jwt->isValid() || empty($this->jwt->getClaims())) {
```

Comment:
Не вижу смысла в проверки наличия клаймов, т.к. `isValid` само по себе проверяет наличие клаймов `iss` и `aud` а если они не сконфигурированы, то такое приложение принимает любой `jwt` токен
```suggestion
        if (!$this->jwt->isValid()) {
```